tests/extension/khr_default_context/default_context.cpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Copyright (c) 2024 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+
+namespace default_context::tests {
+
+#ifdef SYCL_KHR_DEFAULT_CONTEXT
+
+static void testDefaultContext() {
+  sycl::platform platform{};
+  sycl::context defaultContext = platform.khr_get_default_context();
+  CHECK(defaultContext.get_devices() == platform.get_devices());
+}
```

Comment:
If we decide that trying to compile a KHR extension test for a SYCL implementation that does not implement said extension should result in an error, I'd prefer the tests to use a more idiomatic style, e.g.:

```suggestion
TEST_CASE("The default context contains all of the platform's devices") {
  sycl::platform platform{};
  sycl::context defaultContext = platform.khr_get_default_context();
  CHECK(defaultContext.get_devices() == platform.get_devices());
}
```